File: Leetcode/JS/Leetcode89-格雷码.js
var grayCode = function(n) {
    const ans = [0];
    while(n--){
        for(let i = ans.length - 1;i >= 0;i--){
            ans[i] *= 2;
            ans.push(ans[i] + 1);
        }
    }
    return ans;
};

File: Leetcode/Rust/Leetcode89-格雷码.rs
impl Solution {
    pub fn gray_code(mut n: i32) -> Vec<i32> {
        let mut ans = vec![0;1];
        while n != 0 {
            let t = ans.len();
            for i in (0..t).rev(){
                ans[i] *= 2;
                ans.push(ans[i] + 1);
            }
            n -= 1;
        }
        ans
    }
}